### ISSUE TITLE
WithUnwrapOption(false) has no effect (None still serializes to null)

### ISSUE THREAD

**Issue by cmeeren** [Tarmil/FSharp.SystemTextJson#171]
When using ` WithUnwrapOption(false)`, `None` still serializes to `null`.

(I am serializing for logging/assertion/debugging purposes and  want it to serialize to the string "None", as I expect is the intended functionality with `WithUnwrapOption(false)`.)

```
open System.Text.Json
open System.Text.Json.Serialization
open Xunit

let options = JsonSerializerOptions()
options.Converters.Add(
    JsonFSharpOptions
        .Default()
        .WithUnwrapOption(false)
    |> JsonFSharpConverter
)
Assert.Equal(JsonSerializer.Serialize(None), "\"None\"")
```

This fails with:

```
Xunit.Sdk.EqualException
Assert.Equal() Failure
          ↓ (pos 0)
Expected: null
Actual:   "None"
          ↑ (pos 0)
```

**Comment by cmeeren** [comment]
Could this be related to `Option<_>` using `[<CompilationRepresentation(CompilationRepresentationFlags.UseNullAsTrueValue)>]`?

**Comment by Tarmil** [comment]
Yes, it is on purpose that None and ValueNone are serialized as null. I am open to adding an option to customize this though!

**Comment by cmeeren** [comment]
Would be great! I am not serializing for idiomatic JSON, but for displaying structures of F# data (in a better way than `%A`), so displaying `None` instead of `null` would help my users.

**Comment by cmeeren** [comment]
Note by the way that the docs for `WithUnwrapOption` are misleading:

https://github.com/Tarmil/FSharp.SystemTextJson/blob/486924448ff384a6121ec6141df61bd619f9e520/src/FSharp.SystemTextJson/Options.fs#L368-L372

It says that if set, `None` is serialized as `null`. While this technically doesn't say that if it's _not_ set, then `None` is _not_ serialized as `null`, it does give a strong impression that that's the intended meaning.

Also relevant here:

https://github.com/Tarmil/FSharp.SystemTextJson/blob/486924448ff384a6121ec6141df61bd619f9e520/src/FSharp.SystemTextJson/Options.fs#L56-L59

**Comment by cmeeren** [comment]
Hi! Just wondering if it would be possible to implement this soon-ish (say, during the coming week or two)? I need to decide whether to work around this, but the point is moot if this could be supported upstream. 🙂
